gengokit/httptransport/httptransport.go

```diff
@@ -101,6 +102,24 @@ func NewBinding(i int, meth *deftree.ServiceMethod) *Binding {
 		nField.LowCamelName = LowCamelName(nField.Name)
 
 		nBinding.Fields = append(nBinding.Fields, &nField)
+
+		// Emit warnings for certain cases
+		if !nField.IsBaseType {
+			log.Warnf("%s.%s is a custom type '%s', only base types and repeated base "+
+				"types are supported. As a result, the generated HTTP "+
+				"transport will fail to compile. Remove non-base types.",
+				meth.GetName(),
```

Comment:
Oh, yes, I see, the first param is on the same line as the function call. I hadn't noticed that. When making multi-line calls, prefer placing the first parameter on the line after the function call. Just affirming what you said you intend to do.